### ISSUE TITLE
Australian emails going to spam

### ISSUE THREAD

**Issue by JenSheridan** [openfoodfoundation/openfoodnetwork#6529]
## Description
On Saturday order confirmations started not going through:
![image](https://user-images.githubusercontent.com/39814473/102044447-f0892380-3e2a-11eb-84f0-fcf52aee216d.png)
@sauloperez  investigated and found that someone had unchecked 'enable mail deliver' in Aus platform configuration. 
![image](https://user-images.githubusercontent.com/39814473/102044536-1b737780-3e2b-11eb-8bd9-ea05c1f0868a.png)

He did did irb(main):002:0> Spree::Config[:enable_mail_delivery] = true and restart Unicorn and DJ and said the issue seemed over. 

Since then all Australian emails have been going to spam. In some cases people's spam filters aren't even letting it get to the spam folder. 

There were a chunk of emails when the spam first started like this: 
"This is the mail system at host localhost.localdomain.

I'm sorry to have to inform you that your message could not
be delivered to one or more recipients. It's attached below.

For further assistance, please send mail to postmaster.

If you do so, please include this problem report. You can
delete your own text from the attached returned message.

                   The mail system

<recipient email here, I've removed for privacy>: host box.tuilder.com[107.191.57.51] said: 554
    5.7.1 Service unavailable; Client host [43.239.97.238] blocked using
    zen.spamhaus.org; https://www.spamhaus.org/query/ip/43.239.97.238 (in reply
    to RCPT TO command)"

And then for those customer who aren't even getting the emails making it to their spam folder we're getting this message also. 

This is Severity 1 bug for aus as emails aren't getting to customers or enterprises.

**Comment by mkllnk** [comment]
Nobody unticked the email delivery box. The server was provisioned without our email settings. Enabling that tickbox started to deliver all emails to `localhost` which tried to send those emails out but is not authorised. The emails should have gone through Gmail.

**Comment by mkllnk** [comment]
The Spree config is not in sync with our environment:
```
irb(main):002:0> ENV.fetch("MAIL_HOST")
=> "smtp.gmail.com"
irb(main):003:0> Spree::Config[:mail_host]
=> "localhost"
```

**Comment by mkllnk** [comment]
Since I don't know where the configuration went wrong, I manually reset the mail settings in the rails console:
```
$ (cd ~/apps/openfoodnetwork/current/ && bundle exec rails console)
Spree::Config[:mail_host] = "smtp.gmail.com"
Spree::Config[:smtp_username] = "..."
Spree::Config[:smtp_password] = "..."
Spree::Config[:secure_connection_type] = "TLS"
Spree::Config[:mail_port] = "587"
Spree::Config[:mail_auth_type] = "login"
```

I restarted unicorn and delayed job. Emails seem to be delivered again. But we need to be careful with the next deployment.

**Comment by mkllnk** [comment]
One thing that happened on Saturday was a server restart. The whole VPS was restarted by our hosting provider for security updates. We should verify that server restarts persist the email configuration.

**Comment by sauloperez** [comment]
we need to check if it's only the mail configuration that got lost but all or some other ENV vars.

**Comment by mkllnk** [comment]
There are several ways the email configuration can be changed:

1. On every deployment, we call `rake db:seed` which reads environment variables and sets the email config. The environment variables are there and we didn't deploy on Saturday when this happened. So the only way this could have happened would be a dev running the seed script manually with a different environment.

2. Database entries in spree_preferences got overridden. Preferences seem to be all good. Many haven't been changed for a while. `mail_host` was changed on Monday by me but the creation date is over a year old. That proves that entries have not been deleted. A database backup from Friday proves that the config was correct then. And a database backup from Sunday, after the incident, still shows that the database wasn't changed. This rules out option 1 above as well.

3. The cache could have been changed, either in memcached or in the app's memory. I changed my local dev config to use mem_cache_store while memcached was shut down. After booting the console, I get:
  ```
  irb(main):002:0> Spree::Config.mail_host
Cache read: /spree/app_configuration/mail_host ({:value_max_bytes=>4194304})
down_retry_delay not reached for localhost:11211 (14.228 seconds left)
DalliError (No server available): No server available
Cache write: /spree/app_configuration/mail_host ({:value_max_bytes=>4194304})
down_retry_delay not reached for localhost:11211 (14.227 seconds left)
DalliError (No server available): No server available
=> "localhost"
  ```
  So instead of looking in the database, we use the default values if memcached is down. I described in https://github.com/openfoodfoundation/ofn-install/issues/680 how unicorn and delayed_job can boot before memcached which would explain this happening after a server restart. I submitted a pull request to fix the boot order but it's still bad that we don't use the values from the database if the caching service is down.
